Mathlib/Analysis/Normed/Operator/Extend.lean

```diff
@@ -116,3 +116,126 @@ end NormedField
 end Extend
 
 end ContinuousLinearMap
+
+namespace LinearMap
+
+section LeftInverse
+
+variable [DivisionRing 𝕜] [AddCommGroup E] [AddCommGroup F] [Module 𝕜 E] [Module 𝕜 F]
+
+variable (f : E →ₗ[𝕜] F)
+
+open scoped Classical in
+/-- The left inverse of `f : E →ₗ[𝕜] F`.
+
+If `f` is not injective, then we use the junk value `0`. -/
+def leftInverse : F →ₗ[𝕜] E :=
+  if h_inj : LinearMap.ker f = ⊥ then
```

Comment:
Do you have a good sense for which spelling is better from a utilitarian point of view? That is, `LinearMap.ker f = ⊥` or `Function.Injective ⇑f`? My thinking would be the latter as we (did / are planning to) make `Function.Injective` a `fun_prop` goal.